hack/scripts/validate_state.sh

```diff
@@ -0,0 +1,105 @@
+#!/bin/bash
```

Comment:
it may be late to bring this up, but...
bash has a high barrier to entry and isn't really suited to any complex control flow. is there a reason you scripted this instead of adding it to the Go tests in `/test/integration`, or copying that pattern?

Reply:
I agree to this and we thought of going that path. 
We have a task to add it as Go tests but since we started this PR when we were going for GA and based on the timeline the effort seemed less as a bash script.

I'll check with tamilmani if we should work on the pipeline as go tests now since GA has been pushed.
Thanks for bringing it up.

Reply:
@rbtr So had a chat with tamilmani. We have a task for adding the tests for CNI as go tests as mentioned and we will be working on them in the April sprint. 
If it is not a hard blocker, can move ahead with the bash script approach ?

Reply:
not a blocker